### DISCUSSION TITLE
Using search params + disable JS = page not loading (app router)

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#49842] [App Router]
As soon as I use **URL search params** anywhere in my code, all pages up to this component stop loading if JavaScript is disabled in the browser. I either see an empty body, or get stuck on `loading.tsx`.

This happens no matter if I use the server-side `searchParams` prop or the client-side `useSearchParams` hook.

For example, this page loads data from the server and reads the current page from the URL `page` param:

    interface PageProps {
        searchParams: { [key: string]: string | string[] | undefined };
    };
    
    export default async function Page({ searchParams }: PageProps) {
        const page = parseInt(searchParams.page?.toString() || "1");
    
        const data = await BlogApi.getBlogPosts(page);
    
        return <BlogPage data={data} />
    }

Without JavaScript, this page gets stuck on `loading.tsx`. The problem only goes away if I remove `searchParams` completely.

What am I missing? I understand that search params cause **dynamic** rendering. But I should still get data from the server without JavaScript.

**Comment by waldothedeveloper** [comment]
Hey @codinginflow, this has been unanswered for a year almost, did you ever figured it out?

**Reply by codinginflow** [reply]
Yea, I think you have to wrap the component that uses searchParams into a `Suspense` boundary.
